pkg/snmp/traps/formatter.go

```diff
@@ -161,6 +161,28 @@ func NormalizeOID(value string) string {
 	return strings.TrimLeft(value, ".")
 }
 
+// parseValue checks to see if the variable has a mapping in an enum and
+// returns the mapping if it exists, otherwise returns the value unchanged
+func parseValue(variable trapVariable, varMetadata VariableMetadata) interface{} {
+	if variable.VarType == "integer" {
```

Comment:
A couple of suggestions here:
- I would check first for `len(varMetadata.Enumeration) > 0`, IMO it's what we're really checking for here, right?
- I would also ignore the check of VaryType. If we know this variables has an enumeration, we should then try to cast its value to an int even if the VarType is "string" (maybe a mistake in how the device is sending traps?).

So pseudo-code like:

```
1. Check enumeration, if empty returns the var value unchanged.
2. Check cast to int, if fail return the var value unchanged.
3. Return the enumerated value
```